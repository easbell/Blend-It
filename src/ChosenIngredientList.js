import React from 'react';

export default function ChosenIngredientList(props){
  return(
    <div className="chosen-ingredients">
      <h3>Selected Ingredients: </h3>
        {
          props.ingredients.map(chosen => {
            return (
              <p>{chosen}</p>
              )
          })
        }
        <div>
          <button className="button" onClick={props.resetButton}>Clear Selections</button>
        </div>
    </div>
  )
}