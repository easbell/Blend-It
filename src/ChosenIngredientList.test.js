import React from 'react';
import ChosenIngredientList from './ChosenIngredientList.js';
import { shallow } from 'enzyme';

const chosenMock = ['Banana'];
const resetMockFunc = jest.fn();

describe('ChosenIngredientList', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ChosenIngredientList  
        ingredients={chosenMock}
        resetButton={resetMockFunc}
      />
    )
  });

  it('should match a snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should fire off function when clicked', () => {
    wrapper.find('.button').simulate('click');
    expect(resetMockFunc).toBeCalled();
  });
});