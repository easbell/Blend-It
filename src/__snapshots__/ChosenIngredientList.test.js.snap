// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChosenIngredientList should match a snapshot with all data passed in 1`] = `
<div
  className="chosen-ingredients"
>
  <h3>
    Selected Ingredients: 
  </h3>
  <p>
    Banana
  </p>
  <div>
    <button
      className="button"
      onClick={[MockFunction]}
    >
      Clear Selections
    </button>
  </div>
</div>
`;
